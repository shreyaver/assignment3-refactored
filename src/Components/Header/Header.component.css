header {
    min-height: 110px;
    display: flex;
    flex-direction: row;
    align-items: center;
    outline: none;
}
.headerLine {
    flex: 1;
}
.headerLine hr {
    border: none;
    height: 4px;
    color: var(--white); 
    background-color: var(--white);
}
.headerTitle {
    height: 90px;
    width: 250px;
    display: flex;
    flex-direction: column;
    align-items: center;
}
.headerCaption {
    font-size: 1.5rem;
    color: var(--white);
    font-weight: bold;
    font-style: italic;
    text-align: center;
    margin-top: 10px;
}
.circle {
    width: 50px;
    line-height: 50px;
    height: 50px;
    border: 2px solid var(--white);
    border-radius: 50%;
    vertical-align: middle;
    text-align: center;
    font-size: 2rem;
    color: var(--white);
}