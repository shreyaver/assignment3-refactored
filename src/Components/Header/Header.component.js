import React from 'react';
import PropTypes from 'prop-types';
import './Header.component.css';

const Header = (props) => {
  const { handleHeaderClick } = props;
  return (
    <header role="button" onClick={handleHeaderClick} onKeyDown={handleHeaderClick} tabIndex={0}>
      <div className="headerLine">
        <hr />
      </div>
      <div className="headerTitle">
        <div className="circle">
        BS
        </div>
        <div className="headerCaption">
        The Book Shelf
        </div>
      </div>
      <div className="headerLine">
        <hr />
      </div>
    </header>
  );
};

Header.propTypes = {
  handleHeaderClick: PropTypes.func.isRequired,
};

export default Header;
