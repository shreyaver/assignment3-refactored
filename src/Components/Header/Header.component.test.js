import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from './Header.component';

describe('Header', () => {
  it('renders without crashing', () => {
    const header = renderer.create(<Header handleHeaderClick={jest.fn()} />).toJSON();
    expect(header).toMatchSnapshot();
  });
  it('calls function passed as prop when clicked', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<Header handleHeaderClick={mockFunction} />);
    expect(mockFunction).not.toHaveBeenCalled();
    wrapper.find('header').simulate('click');
    expect(mockFunction).toHaveBeenCalled();
  });
});
