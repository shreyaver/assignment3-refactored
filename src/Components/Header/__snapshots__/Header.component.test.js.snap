// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header
  onClick={[MockFunction]}
  onKeyDown={[MockFunction]}
  role="button"
  tabIndex={0}
>
  <div
    className="headerLine"
  >
    <hr />
  </div>
  <div
    className="headerTitle"
  >
    <div
      className="circle"
    >
      BS
    </div>
    <div
      className="headerCaption"
    >
      The Book Shelf
    </div>
  </div>
  <div
    className="headerLine"
  >
    <hr />
  </div>
</header>
`;
